### PR TITLE
Increase Character Slots from 9 to 12

### PR DESCRIPTION
**Pull request by Jedzkie**
- kRO Mains recently increased their character slots from 11 to 12.

---
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1184)

### CODE REVIEW

**Comment by hemagx** [comment]
i remember old clients need to be hexed for this.
can someone check all client limits in old clients ? this one may need to be put under PACKETVER condition

**Comment by Jedzkie** [comment]
@hemagx Please do check.

**Comment by dastgir** [comment]
Nice +1

**Comment by Travesso** [comment]
Nice +1

**Comment by Jedzkie** [comment]
Any news on this @hemagx ?

File: src/common/mmo.h [hunk 1]
```diff
@@ -114,7 +114,11 @@
 
 #define MAX_INVENTORY 100
 //Max number of characters per account. Note that changing this setting alone is not enough if the client is not hexed to support more characters as well.
-#define MAX_CHARS 9
+#if PACKETVER < 20090617
```
Comment (hemagx) [inline src/common/mmo.h]: sure it's < and not <= ?

**Comment by EPuncker** [comment]
:+1:

**Comment by Ridley8819** [comment]
<= 20100407

**Comment by Ridley8819** [comment]
@4144 @MishimaHaruna this can be merged

**Comment by 4144** [comment]
Not sure is this correct for this version or not,
but while it changed, better add before line:
<code>#ifndef MAX_CHARS</code>
and after line:
<code>#endif</code>

**Comment by Asheraf** [comment]
@4144

**Comment by 4144** [comment]
@Asheraf code looks fine, but i dont know is this change correct for this clients.
Need some one else to confirm this.

**Comment by Asheraf** [comment]
Me and ridley have test it, clients before that date need to be diffed to support 12 chars also the diff for increasing char slots got removed after this date.

in 2010-04-07aRagexeRE:
https://github.com/MStr3am/roclientside/blob/master/Diff_Project/kRO/RagexeRE/2010/2010-04-07/2010-04-07aRagexeRE.diff#L1427-L1437

in 2010-04-14eRagexeRE:
https://github.com/MStr3am/roclientside/blob/master/Diff_Project/kRO/RagexeRE/2010/2010-04-14/2010-04-14eRagexeRE.diff

**Comment by MishimaHaruna** [comment]
This packetver check should be kept consistent with the check in `char_mmo_send_characters()`, since it's closely related. That function says that for any clients `>= 20100413` we send the amount of available slots to the client (which matches what you observed in the client diffs, where they removed the slot modification, since starting with this date it's no longer hardcoded)

While I don't know if a 2010-04-13 client exists, and if it doesn't, this won't make any difference, I believe that if `char_mmo_send_characters()` uses 2010-04-13 as a threshold, then this patch also should. Otherwise both should be changed to use 2010-04-07.

**Comment by Asheraf** [comment]
@MishimaHaruna done changed to check 20100413.

**Comment by MishimaHaruna** [comment]
Thank you, merging

**Pull request merged**